pkg/util/cfg/dynamic.go

```diff
@@ -22,14 +22,18 @@ func DynamicUnmarshal(dst DynamicCloneable, args []string, fs *flag.FlagSet) err
 		// Next populate the config from the config file, we do this to populate the `common`
 		// section of the config file by taking advantage of the code in ConfigFileLoader which will load
 		// and process the config file.
-		ConfigFileLoader(args, "config.file"),
+		ConfigFileLoader(args, "config.file", true),
 		// Apply any dynamic logic to set other defaults in the config. This function is called after parsing the
 		// config files so that values from a common, or shared, section can be used in
 		// the dynamic evaluation
 		dst.ApplyDynamicConfig(),
 		// Load configs from the config file a second time, this will supersede anything set by the common
 		// config with values specified in the config file.
-		ConfigFileLoader(args, "config.file"),
+		// By loading the config file twice and unmarshaling it into the same object,
+		// using strict yaml unmarshal causes an `already set in map` error with the `Clients` config,
+		// because it's a map that already has the keys we are trying to unmarshal into it.
+		// That is why we don't use strict for the second marshaling.
+		ConfigFileLoader(args, "config.file", false),
```

Comment:
Thanks for the explanation but I'd love it if we could figure out how to solve this without changing this.
I'm concerned making this unstrict will introduce some subtle bugs.
I don't know this part of the code well, which is contributing to my skepticism.